src/adapters/node.ts

```diff
@@ -1,32 +1,24 @@
 import http from "node:http"
-import {once} from "node:events"
+import { transformToNodeBuilder } from "src/edge-runtime/transform-to-node"
 
 export const startServer = (edgeSpec: any, port?: number) => {
-  const server = http.createServer(async (req, res) => {
-    const chunks: Uint8Array[] = []
-    req.on("data", (chunk) => chunks.push(chunk))
-    await once(req, "end")
-    const body = Buffer.concat(chunks)
-
-    const fullUrl = `http://localhost:${req.socket.localPort}${req.url}`
-    const fetchRequest = new Request(fullUrl, {
-      method: req.method,
-        headers: Object.entries(req.headers) as [string, string][],
-        body: req.method === "GET" ? undefined : body,
-    })
+  const transformToNode = transformToNodeBuilder({
+    defaultOrigin: `http://localhost${port ? `:${port}` : ""}`,
+  })
 
-    const {matchedRoute, routeParams} = edgeSpec.routeMatcher(new URL(fetchRequest.url).pathname)
-    const handler = edgeSpec.routeMapWithHandlers[matchedRoute]
-    fetchRequest.pathParams = routeParams
+  const server = http.createServer(
+    transformToNode(async (fetchRequest) => {
+      const { matchedRoute, routeParams } = edgeSpec.routeMatcher(
+        new URL(fetchRequest.url).pathname
+      )
+      const handler = edgeSpec.routeMapWithHandlers[matchedRoute]
+      ;(fetchRequest as Request & { pathParams?: any }).pathParams = routeParams
 
-    const fetchResponse: Response = await handler(fetchRequest)
+      const fetchResponse: Response = await handler(fetchRequest)
 
-    res.statusCode = fetchResponse.status
-    fetchResponse.headers.forEach((value, key) => {
-      res.setHeader(key, value)
+      return fetchResponse
     })
-    res.end(await fetchResponse.text())
-  })
+  )
```

Comment:
this should have been documented better, but we don't want to run within the edge runtime here--this is for when you're targeting Node, either because you want to use Node APIs or you're unable to use the edge runtime (I think you might not be allowed to use it in Vercel endpoints)

Reply:
Ah, I see! I'll revert this in a future pr 👍